homeassistant/scripts/check_config.py

```diff
@@ -264,9 +265,17 @@ def mock_logger_exception_bootstrap(msg, *params):
     yaml.yaml.SafeLoader.add_constructor('!secret', yaml._secret_yaml)
 
     try:
-        with patch('homeassistant.util.logging.AsyncHandler._process'):
+        if new_method:
+            hass = core.HomeAssistant()
```

Comment:
Why would we even need to instantiate `hass` to check configuration? That is not necessary.

Reply:
Indeed, only need loop and config

Reply:
Why would you need loop or config? You really only need config path.

You only need `loop` if you plan on using `await`, but just validating some dicts doesn't require any awaits.

Reply:
I was thinking `load_yaml_config_file` should actually be awaited, sice it does IO? 
But otherwise there is no IO (except for importing all the plaforms, which I would think also needs awaiting)